# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/d3d12/src/debug.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/d3d12/src/debug.rs b/third_party/rust/d3d12/src/debug.rs
--- a/third_party/rust/d3d12/src/debug.rs
+++ b/third_party/rust/d3d12/src/debug.rs
@@ -1,43 +1,43 @@
-use crate::com::WeakPtr;
-use winapi::um::d3d12sdklayers;
-#[cfg(any(feature = "libloading", feature = "implicit-link"))]
-use winapi::Interface as _;
-
-pub type Debug = WeakPtr<d3d12sdklayers::ID3D12Debug>;
-
-#[cfg(feature = "libloading")]
-impl crate::D3D12Lib {
-    pub fn get_debug_interface(&self) -> Result<crate::D3DResult<Debug>, libloading::Error> {
-        type Fun = extern "system" fn(
-            winapi::shared::guiddef::REFIID,
-            *mut *mut winapi::ctypes::c_void,
-        ) -> crate::HRESULT;
-
-        let mut debug = Debug::null();
-        let hr = unsafe {
-            let func: libloading::Symbol<Fun> = self.lib.get(b"D3D12GetDebugInterface")?;
-            func(&d3d12sdklayers::ID3D12Debug::uuidof(), debug.mut_void())
-        };
-
-        Ok((debug, hr))
-    }
-}
-
-impl Debug {
-    #[cfg(feature = "implicit-link")]
-    pub fn get_interface() -> crate::D3DResult<Self> {
-        let mut debug = Debug::null();
-        let hr = unsafe {
-            winapi::um::d3d12::D3D12GetDebugInterface(
-                &d3d12sdklayers::ID3D12Debug::uuidof(),
-                debug.mut_void(),
-            )
-        };
-
-        (debug, hr)
-    }
-
-    pub fn enable_layer(&self) {
-        unsafe { self.EnableDebugLayer() }
-    }
-}
+use crate::com::WeakPtr;
+use winapi::um::d3d12sdklayers;
+#[cfg(any(feature = "libloading", feature = "implicit-link"))]
+use winapi::Interface as _;
+
+pub type Debug = WeakPtr<d3d12sdklayers::ID3D12Debug>;
+
+#[cfg(feature = "libloading")]
+impl crate::D3D12Lib {
+    pub fn get_debug_interface(&self) -> Result<crate::D3DResult<Debug>, libloading::Error> {
+        type Fun = extern "system" fn(
+            winapi::shared::guiddef::REFIID,
+            *mut *mut winapi::ctypes::c_void,
+        ) -> crate::HRESULT;
+
+        let mut debug = Debug::null();
+        let hr = unsafe {
+            let func: libloading::Symbol<Fun> = self.lib.get(b"D3D12GetDebugInterface")?;
+            func(&d3d12sdklayers::ID3D12Debug::uuidof(), debug.mut_void())
+        };
+
+        Ok((debug, hr))
+    }
+}
+
+impl Debug {
+    #[cfg(feature = "implicit-link")]
+    pub fn get_interface() -> crate::D3DResult<Self> {
+        let mut debug = Debug::null();
+        let hr = unsafe {
+            winapi::um::d3d12::D3D12GetDebugInterface(
+                &d3d12sdklayers::ID3D12Debug::uuidof(),
+                debug.mut_void(),
+            )
+        };
+
+        (debug, hr)
+    }
+
+    pub fn enable_layer(&self) {
+        unsafe { self.EnableDebugLayer() }
+    }
+}